consensus/types/src/blob_sidecar.rs

```diff
@@ -185,57 +260,23 @@ impl<T: EthSpec> BlobSidecar<T> {
         // Fixed part
         Self::empty().as_ssz_bytes().len()
     }
-}
 
-#[derive(
-    Debug,
-    Clone,
-    Serialize,
-    Deserialize,
-    Encode,
-    Decode,
-    TreeHash,
-    TestRandom,
-    Derivative,
-    arbitrary::Arbitrary,
-)]
-#[derivative(PartialEq, Eq, Hash)]
-pub struct BlindedBlobSidecar {
-    pub block_root: Hash256,
-    #[serde(with = "serde_utils::quoted_u64")]
-    pub index: u64,
-    pub slot: Slot,
-    pub block_parent_root: Hash256,
-    #[serde(with = "serde_utils::quoted_u64")]
-    pub proposer_index: u64,
-    pub blob_root: Hash256,
-    pub kzg_commitment: KzgCommitment,
-    pub kzg_proof: KzgProof,
-}
-
-impl BlindedBlobSidecar {
-    pub fn empty() -> Self {
-        Self {
-            block_root: Hash256::zero(),
-            index: 0,
-            slot: Slot::new(0),
-            block_parent_root: Hash256::zero(),
-            proposer_index: 0,
-            blob_root: Hash256::zero(),
-            kzg_commitment: KzgCommitment::empty_for_testing(),
-            kzg_proof: KzgProof::empty(),
+    pub fn build_sidecars(
+        blobs: BlobsList<T>,
+        block: &SignedBeaconBlock<T>,
+        kzg_proofs: KzgProofs<T>,
+    ) -> Result<BlobSidecarList<T>, BlobSidecarError> {
+        let mut blob_sidecars = vec![];
+        for (i, (kzg_proof, blob)) in kzg_proofs.iter().zip(blobs).enumerate() {
+            let blob_sidecar = BlobSidecar::new(i, blob, block, *kzg_proof)?;
+            blob_sidecars.push(Arc::new(blob_sidecar));
         }
+        Ok(VariableList::from(blob_sidecars))
     }
 }
 
-impl SignedRoot for BlindedBlobSidecar {}
-
 pub type SidecarList<T, Sidecar> = VariableList<Arc<Sidecar>, <T as EthSpec>::MaxBlobsPerBlock>;
 pub type BlobSidecarList<T> = SidecarList<T, BlobSidecar<T>>;
```

Comment:
We can get rid of the `SidecarList` type alias now we no longer need this generic.